recipes-bsp/firmware-imx/firmware-imx-8_7.8.bb

```diff
@@ -0,0 +1,30 @@
+# Copyright 2018 NXP
+SUMMARY = "Freescale i.MX firmware for 8 family"
+DESCRIPTION = "Freescale i.MX firmware for 8 family"
+
+require firmware-imx-${PV}.inc
+
+inherit deploy
+
+do_install() {
+    install -d ${D}/boot
+    # Cadence HDMI
+    install -m 0644 ${S}/firmware/hdmi/cadence/hdmitxfw.bin ${D}/boot
+    install -m 0644 ${S}/firmware/hdmi/cadence/hdmirxfw.bin ${D}/boot
+    install -m 0644 ${S}/firmware/hdmi/cadence/dpfw.bin ${D}/boot
+    # SECO
+    install -m 0644 ${S}/firmware/seco/mx8qm-ahab-container.img ${D}/boot
+}
+
+do_deploy() {
+    cp ${D}/boot/* ${DEPLOYDIR}
+}
+addtask deploy after do_install before do_build
+
+PACKAGE_ARCH = "${MACHINE_SOCARCH}"
+
+FILES_${PN} = "/boot"
```

Comment:
What decides if something ought to be installed or not is if it is loaded from the filesystem or not. If it is used as binaries to *concatenate* with the bootloader to generate a bitstream, it can be on deploy. If it is loaded from the filesystem, we ought to package it.

Reply:
Installing to /boot is wrong because those files are not used from /boot. They are collected by imx-boot and concatenated.